packages/esm-outpatient-app/src/visit-header/visit-header-side-menu.component.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useRef } from 'react';
+import { LeftNavMenu } from '@openmrs/esm-framework';
+
+interface VisitHeaderSideMenu {
+  isExpanded: boolean;
+  toggleSideMenu: () => void;
+}
+
+const VisitHeaderSideMenu: React.FC<VisitHeaderSideMenu> = ({ isExpanded, toggleSideMenu }) => {
+  const menuRef = useRef(null);
+
+  useEffect(() => {
```

Comment:
Why not [useOnClickOutside](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/docs/API.md#useonclickoutside)?

Reply:
The `useOnClickOutside` hook handles clicking outside the side menu, we also want to close the side menu when the user clicks a link and popstate event happens.

Reply:
```ts
  const menuRef = useOnClickOutside(toggleSideMenu);

  React.useEffect(() => {
    window.addEventListener('popstate', toggleSideMenu);
    return window.addEventListener('popstate', toggleSideMenu);
  }, [toggleSideMenu]);
```

?